### PR TITLE
Implement sovereign extended header request handler

### PR DESCRIPTION
**Pull request by mariusmihaic**
## Reasoning behind the pull request
- Implement sovereign extended header request handler
  
## Proposed changes
- 
- 
- 

## Testing procedure
- 
- 
- 

## Pre-requisites

Based on the [Contributing Guidelines](https://github.com/multiversx/mx-chain-go/blob/master/.github/CONTRIBUTING.md#branches-management) the PR author and the reviewers must check the following requirements are met:
- was the PR targeted to the correct branch?
- if this is a larger feature that probably needs more than one PR, is there a `feat` branch created?
- if this is a `feat` branch merging, do all satellite projects have a proper tag inside `go.mod`?

### CODE REVIEW

File: dataRetriever/requestHandlers/sovereignRequestHandler.go [hunk 1]
```diff
@@ -19,14 +25,95 @@ func NewSovereignResolverRequestHandler(resolverRequestHandler *resolverRequestH
...
+		return
+	}
+
+	srrh.addRequestedItems([][]byte{key}, suffix)
+}
+
+func (srrh *sovereignResolverRequestHandler) getExtendedShardHeaderRequester(_ uint32) (dataRetriever.Requester, error) {
+	headerRequester, err := srrh.requestersFinder.IntraShardRequester(factory.ExtendedHeaderProofTopic)
+	if err != nil {
+		log.Warn("extended header proof container not found, available requesters in container",
```
Comment (SebastianMarian) [inline dataRetriever/requestHandlers/sovereignRequestHandler.go:75]: "getExtendedShardHeaderRequester.IntraShardRequester: extended..."

File: dataRetriever/requestHandlers/sovereignRequestHandler.go [hunk 2]
```diff
@@ -19,14 +25,95 @@ func NewSovereignResolverRequestHandler(resolverRequestHandler *resolverRequestH
...
+		)
+		return
+	}
+
+	srrh.whiteList.Add([][]byte{hash})
+
+	epoch := srrh.getEpoch()
+	err = headerRequester.RequestDataFromHash(hash, epoch)
+	if err != nil {
+		log.Debug("RequestShardHeader.RequestDataFromHash",
```
Comment (SebastianMarian) [inline dataRetriever/requestHandlers/sovereignRequestHandler.go:110]: "RequestExtendedShardHeader.RequestDataFromHash"

File: dataRetriever/requestHandlers/sovereignRequestHandler.go [hunk 3]
```diff
@@ -19,14 +25,95 @@ func NewSovereignResolverRequestHandler(resolverRequestHandler *resolverRequestH
...
+		log.Warn("wrong assertion type when creating header requester")
+		return
+	}
+
+	srrh.whiteList.Add([][]byte{key})
+
+	epoch := srrh.getEpoch()
+	err = headerRequester.RequestDataFromNonce(nonce, epoch)
+	if err != nil {
+		log.Debug("RequestShardHeaderByNonce.RequestDataFromNonce",
```
Comment (SebastianMarian) [inline dataRetriever/requestHandlers/sovereignRequestHandler.go:61]: "RequestExtendedShardHeaderByNonce.RequestDataFromNonce"

File: dataRetriever/requestHandlers/sovereignRequestHandler.go [hunk 4]
```diff
@@ -19,14 +25,95 @@ func NewSovereignResolverRequestHandler(resolverRequestHandler *resolverRequestH
...
+		log.Error("RequestShardHeaderByNonce.getExtendedShardHeaderRequester",
+			"error", err.Error(),
+			"shard", srrh.shardID,
+		)
+		return
+	}
+
+	headerRequester, ok := requester.(NonceRequester)
+	if !ok {
+		log.Warn("wrong assertion type when creating header requester")
```
Comment (sasurobert) [inline dataRetriever/requestHandlers/sovereignRequestHandler.go:52]: this is error.

File: dataRetriever/requestHandlers/sovereignRequestHandler.go [hunk 5]
```diff
@@ -19,14 +25,95 @@ func NewSovereignResolverRequestHandler(resolverRequestHandler *resolverRequestH
...
+	}
+
+	log.Debug("sovereign: requesting extended shard header by nonce from network",
+		"shard", srrh.shardID,
+		"nonce", nonce,
+	)
+
+	requester, err := srrh.getExtendedShardHeaderRequester(srrh.shardID)
+	if err != nil {
+		log.Error("RequestShardHeaderByNonce.getExtendedShardHeaderRequester",
```
[the remark this answers is not in the document]
Comment (mariusmihaic) [reply dataRetriever/requestHandlers/sovereignRequestHandler.go:43]: Right, changed it

File: dataRetriever/requestHandlers/sovereignRequestHandler.go (hunk 4 above)
[the remark this answers is not in the document]
Comment (mariusmihaic) [reply dataRetriever/requestHandlers/sovereignRequestHandler.go:52]: Right, changed it

File: dataRetriever/requestHandlers/sovereignRequestHandler.go (hunk 1 above)
Comment (mariusmihaic) [reply dataRetriever/requestHandlers/sovereignRequestHandler.go:75]: Changed it

File: dataRetriever/requestHandlers/sovereignRequestHandler.go [hunk 6]
```diff
@@ -19,14 +25,95 @@ func NewSovereignResolverRequestHandler(resolverRequestHandler *resolverRequestH
...
 // RequestExtendedShardHeader method asks for extended shard header from the connected peers by nonce
-func (srrh *sovereignResolverRequestHandler) RequestExtendedShardHeader(_ []byte) {
-	//TODO: This method should be implemented for sovereign chain
+func (srrh *sovereignResolverRequestHandler) RequestExtendedShardHeader(hash []byte) {
+	suffix := fmt.Sprintf("%s_%d", uniqueHeadersSuffix, srrh.shardID)
+	if !srrh.testIfRequestIsNeeded(hash, suffix) {
+		return
+	}
+
+	log.Debug("sovereign: requesting extended shard header from network by hash",
```
[the remark this answers is not in the document]
Comment (mariusmihaic) [reply dataRetriever/requestHandlers/sovereignRequestHandler.go:91]: Changed it

File: dataRetriever/requestHandlers/sovereignRequestHandler.go [hunk 7]
```diff
@@ -19,14 +25,95 @@ func NewSovereignResolverRequestHandler(resolverRequestHandler *resolverRequestH
...
+		"shard", srrh.shardID,
+		"hash", hash,
+	)
+
+	headerRequester, err := srrh.getExtendedShardHeaderRequester(srrh.shardID)
+	if err != nil {
+		log.Error("sovereignResolverRequestHandler.getExtendedShardHeaderRequester",
```
[the remark this answers is not in the document]
Comment (mariusmihaic) [reply dataRetriever/requestHandlers/sovereignRequestHandler.go:98]: Changed it

File: dataRetriever/requestHandlers/sovereignRequestHandler.go (hunk 2 above)
Comment (mariusmihaic) [reply dataRetriever/requestHandlers/sovereignRequestHandler.go:110]: Changed it

File: dataRetriever/requestHandlers/sovereignRequestHandler.go (hunk 3 above)
Comment (mariusmihaic) [reply dataRetriever/requestHandlers/sovereignRequestHandler.go:61]: Changed it

File: dataRetriever/requestHandlers/sovereignRequestHandler.go (hunk 4 above)
Comment (mariusmihaic) [reply dataRetriever/requestHandlers/sovereignRequestHandler.go:52]: Changed it to error

**Review by sasurobert — APPROVED** [review]

**Pull request merged**